internal/isoeditor/rhcos.go

```diff
@@ -6,66 +6,94 @@ import (
+	// CustomIsoObjectName is a name pattern for the ISO template
+	CustomIsoObjectName string = "rhcos-%s-minimal.iso"
 )
 
 type Editor interface {
-	CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) error
+	CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) (string, error)
 }
 
 type rhcosEditor struct {
-	isoHandler    Handler
-	objectHandler s3wrapper.API
+	isoHandler    isoutil.Handler
 	rootfsURL     string
+	isoTmpWorkDir string
+	log           logrus.FieldLogger
 }
 
-func NewEditor(isoHandler Handler, objectHandler s3wrapper.API, rootfsURL string) Editor {
+func NewEditor(isoHandler isoutil.Handler, rootfsURL, isoTmpWorkDir string, log logrus.FieldLogger) Editor {
 	return &rhcosEditor{
-		isoHandler:    isoHandler,
-		objectHandler: objectHandler,
-		rootfsURL:     rootfsURL,
+		isoHandler: isoHandler,
+		rootfsURL:  rootfsURL,
+		log:        log,
 	}
 }
 
-func (e *rhcosEditor) CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) error {
+func CreateEditor(isoPath string, rootfsURL string, log logrus.FieldLogger) Editor {
+	isoTmpWorkDir, err := ioutil.TempDir("", BaseIsoTempDir)
+	if err != nil {
+		return nil
+	}
+	isoHandler := isoutil.NewHandler(isoPath, isoTmpWorkDir)
+	return NewEditor(isoHandler, rootfsURL, isoTmpWorkDir, log)
+}
+
+func GetMinimalIsoObject(openshiftVersion string) string {
+	return fmt.Sprintf(CustomIsoObjectName, openshiftVersion)
+}
+
+func GetRootfsURL(serviceBaseURL, openshiftVersion string) string {
+	return fmt.Sprintf("%s/api/assisted-install/v1/boot-files?file_type=rootfs.img&openshift_version=%s",
+		serviceBaseURL, openshiftVersion)
+}
+
+func (e *rhcosEditor) CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) (string, error) {
 	if err := e.isoHandler.Extract(); err != nil {
-		return err
+		return "", err
 	}
+	defer os.Remove(e.isoTmpWorkDir)
 
 	if err := os.Remove(e.isoHandler.ExtractedPath("images/pxeboot/rootfs.img")); err != nil {
-		return err
+		return "", err
 	}
 
 	if err := e.addRootFSURL(); err != nil {
-		return err
+		e.log.WithError(err).Warnf("Can't add rootfs_url (missing cfg files)")
 	}
 
-	objectName := fmt.Sprintf("rhcos-%s.iso", openshiftVersion)
-	if err := e.createAndUpload(ctx, objectName); err != nil {
-		return err
+	// create a temp dir for writing the iso there.
+	outDir, err := ioutil.TempDir("", CustomIsoTempDir)
+	if err != nil {
+		return "", err
 	}
 
-	return nil
-}
-
-func (e *rhcosEditor) createAndUpload(ctx context.Context, objectName string) error {
-	// create a temp dir and write the iso there.
-	outDir, err := ioutil.TempDir("", "customiso")
-	if err != nil {
-		return err
+	objectName := GetMinimalIsoObject(openshiftVersion)
+	e.log.Infof("Creating minimal ISO template: %s/%s", outDir, objectName)
+	if err := e.create(ctx, objectName, outDir); err != nil {
+		return "", err
 	}
-	defer os.RemoveAll(outDir)
 
-	outPath := filepath.Join(outDir, objectName)
+	return outDir, nil
```

Comment:
Instead of returning the output directory maybe we create a new method for the output iso path?
Then we can keep the cache location and/or the final file name an internal detail.

Reply:
sure, cached in rhcosEditor struct.